board.cpp

```diff
@@ -0,0 +1,83 @@
+#include "board.h"
+
+
+Board::Board() {
+	this->board = std::vector<int>(BOARD::TRUE_BOARD_SIZE, 0);
+	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
+	this->currentHeight = 1;
+	
+	// place buffer at beginning and end of each row
+	for (int row = 0, size = BOARD::TRUE_COLUMN_SIZE; row < size; ++row) {
+		this->board[row * BOARD::TRUE_ROW_SIZE] = -1;
+		this->board[row * BOARD::TRUE_ROW_SIZE + BOARD::TRUE_ROW_SIZE - 1] = -1;
+	}
+	for (int col = 0, size = BOARD::TRUE_ROW_SIZE; col < size; ++col) {
+		this->board[col] = -1;
+	}
+}
+
+inline int Board::translateRow(int row) {
+	return std::max(0, row + BOARD::ROW_OFFSET); 
+}
+
+inline int Board::translateCol(int col) {
+	return std::max(0, std::min(col + BOARD::COLUMN_OFFSET, 11)); // returns 0 or 11 if row + offset is not between 1 and 10
+}
```

Comment:
inline functions are great but that also means you are at the mercy of the compiler to perform these function optimization. These functions seems simple enough so maybe consider defining macros for them?